src/plugins/intel_gpu/src/plugin/ops/loop.cpp

```diff
@@ -103,6 +103,13 @@ static void SetLoopInputOutputMap(ProgramBuilder& p,
             cldnn::primitive_id from_id = layer_type_name_ID(from);
 
             back_edges_maps.emplace_back(from_id, to_id);
+            if (std::dynamic_pointer_cast<ov::op::v0::Result>(from) != nullptr && std::dynamic_pointer_cast<ov::op::v0::Parameter>(to) != nullptr) {
+                if (to->get_partial_shape() != from->get_input_partial_shape(0)) {
+                    if (to->get_partial_shape().is_static() && !from->get_input_partial_shape(0).is_static()) {
+                        to->set_partial_shape(from->get_input_partial_shape(0));
+                    }
+                }
+            }
```

Comment:
Could you please my trial commits (https://github.com/ahnyoung-paul/openvino/commit/7137836733d75557a8b2618eae1002c8cc2cd077) ? This code works without setting the layout to dynamic in ops/loops.cpp. GPU plugin don't recommend changing static shapes to dynamic shapes like this. Please update the code after refer this commit.

I verified it is working for the issued tests. thanks.